### DISCUSSION TITLE
Detect and ban spam bots in live stream chats

### DISCUSSION THREAD

**Discussion by DoRullings** [ThioJoe/YT-Spammer-Purge#838] [Ideas]
I've noticed that some YouTubers struggle with spam chat bots while live streaming. I've never used YouTube's API other than some very small projects purely for fun, and I don't know if the API allow this, but if it is possible to kick/ban chat bots that obviously spamming would be a feature I believe many will appreciate. I got the idea while watching [The DUI Guy+](https://www.youtube.com/c/TheDUIGuy) live stream [this morning](https://youtu.be/aDSs6e4lJms)  were he told that he had to close the chat for everyone except paying members because there had been too many spam bots in the chat the day before. He had four moderators and they still couldn't keep up with all the spam bots. He is a lawyer, but even so I honestly don't believe he closed the chat for non-members to lure his viewers into being paying members. This [link](https://youtu.be/aDSs6e4lJms?t=26) will take you to the point were he talks about how frustrating this has been. I'll take a look into it myself, but I'm not a YouTuber and I don't know how to test this. 

What do you think? Is it possible? If yes, should it be a feature in this app or an separate app? I would think that the spam detection in this app could be adapted to a chat stream, but the app must be running continuously to analyse the chat stream.

**Comment by TechStudent10** [comment]
I'm not sure if it's easy to just hook up the existing script and change some stuffs to make it livestream compatible. Maybe it is, but I haven't seen tons of YouTube livestream bots other than Nightbot, which is a YouTube Official bot. Would love to be proven wrong, though.

**Comment by SilverscaleClockwork** [comment]
It would be possible to write a bot with the purpose of filtering livestream chats, since the Youtube Api allows for such things.

NightBot is by far not the only bot for youtube streams.
The webpage of the bot says that Nightbot was created by NightDev 
(Some other sources say that it was created by Kona but it's hard to get anything about that.)
 
The only problem right now is that the project isn't that modular that it would be possible to copy every detection mechanism into a bot.
Writing a bot in itself is simple, the problem is though that it would quickly get rate limited...

Rate limiting is something that stops a bot from sending too many requests.
In case of comments, you can pull multiple comments at once.
A stream tho has live chat, meaning you can't bulk request.

So if you are able to prevent rate limiting somehow, you can make a bot.
